packages/block/src/header.ts

```diff
@@ -65,6 +65,9 @@ export class BlockHeader {
 
   public readonly common: Common
 
+  private keccakFunction?: Function
+  private static keccakFunction: ((msg: Uint8Array) => Uint8Array) | undefined
```

Comment:
Ok, all this weird static/not-static doubling should really go away. Not sure why this is happening, but this should really be only the member property like:

```typescript
protected keccakFunction: ((msg: Uint8Array) => Uint8Array)
```

Then all `!` TypeScript operators should also be not necessary.

Side thing: can you also please leave make this `protected` instead of `private`? This leaves more room for community users who want to subclass the `BlockHeader` class (so `Block` the same). Would be nice if you can also update the other member declarations (e.g. in tx) for `keccakFunction` along this line.